expression/builtin_time.go

```diff
@@ -4785,15 +4789,15 @@ func (b *builtinPeriodAddSig) evalInt(row chunk.Row) (int64, bool, error) {
 		return 0, true, err
 	}
 
-	if p == 0 {
-		return 0, false, nil
-	}
-
 	n, isNull, err := b.args[1].EvalInt(b.ctx, row)
 	if isNull || err != nil {
 		return 0, true, err
 	}
 
+	if !validPeriod(p) {
```

Comment:
should we validate this argument before `b.args[1].EvalInt(b.ctx, row)`?

Reply:
No... in MySQL, if `p` is invalid but `n` is `NULL`, the result is `NULL`, so we have to check if `n` is `NULL` first.

Reply:
Maybe we also need to add this explanation as a comment.

Reply:
Comment is added.